scylla/src/transport/session.rs

```diff
@@ -1140,6 +1140,17 @@ impl Session {
         Ok(Some(tracing_info))
     }
 
+    // Returns which replicas are likely to take part in handling the query.
+    // If a list of replicas cannot be easily narrowed down, all available replicas
+    // will be returned.
+    pub fn estimate_replicas_for_query(&self, statement: &Statement) -> Vec<Arc<Node>> {
```

Comment:
If we can't compute the set of replicas, maybe just returning `None` would make more sense?

Reply:
I considered it, but my reasoning is that it returns the narrowest possible set of replicas that will take part in this query. E.g. for `SELECT * FROM table`, our best guess is "all replicas", same for any range query. And point queries usually get an exact match.